### PR TITLE
Acceptance scenarios for DTD-397 & Refactor existing tests

### PR DESCRIPTION
**Pull request by byteray**

### CODE REVIEW

File: src/test/resources/features/ifs/instalment_calculation/InstalmemtCalculationSingleDebt.feature [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      | debtId | debtAmount | mainTrans | subTrans |
+      | 1234   | 100000     | 1545      | 1000     |
+    When the instalment calculation detail is sent to the ifs service
+    Then IFS response contains expected values
+      | instalmentNumber | daysAfterToday | paymentFrequency | frequencyPassed | amountDue | instalmentBalance | interestRate | expectedNumberOfInstalments |
+      | 1                | 1              | monthly          | 0               | 4270      | 100000            | 2.6          | 24                          |
+
+# DTD-397 Edge-cases below
+
+  Scenario: Should return an error from IFS if duration is not provided and quote type is duration
```
Comment (rogerball) [inline src/test/resources/features/ifs/instalment_calculation/InstalmemtCalculationSingleDebt.feature]: Should be other way round, I believe? eg ...
quote type = instalment amount then Frequency and duration are mandatory fields.
quote type = duration, then frequency and instalment amount are mandatory fields .

File: src/test/resources/features/ifs/instalment_calculation/InstalmemtCalculationSingleDebt.feature [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    And the instalment calculation has no postcodes
+    And no initial payment for the debt item charge
+    And the instalment calculation has debt item charges
+      | debtId | debtAmount | mainTrans | subTrans |
+      | 1234   | 100000     | 1545      | 1000     |
+    When the instalment calculation detail is sent to the ifs service
+    Then Ifs service returns response code 400
+    And Ifs service returns error message {"statusCode":400,"reason":"Invalid JSON error from IFS","message":"For a quote type of `duration`, the duration field is required"}
+
+  Scenario: Should return an error from IFS if quote type is instalment amount but duration is provided
```
Comment (rogerball) [inline src/test/resources/features/ifs/instalment_calculation/InstalmemtCalculationSingleDebt.feature]: Should be other way round, I believe? eg ...
quote type = instalment amount then Frequency and duration are mandatory fields.
quote type = duration, then frequency and instalment amount are mandatory fields .

Comment (byteray) [reply src/test/resources/features/ifs/instalment_calculation/InstalmemtCalculationSingleDebt.feature]: @Jxr227 Please confirm this is true, Just had a convo with Roger now and he thinks we need to discuss and confirm with Helen

File: src/test/resources/features/ifs/instalment_calculation/InstalmemtCalculationSingleDebt.feature (hunk 1 above)
Comment (byteray) [reply src/test/resources/features/ifs/instalment_calculation/InstalmemtCalculationSingleDebt.feature]: @Jxr227 Please confirm this is true, Just had a convo with Roger now and he thinks we need to discuss and confirm with Helen

Comment (Jxr227) [reply src/test/resources/features/ifs/instalment_calculation/InstalmemtCalculationSingleDebt.feature]: My understnading is quoteType = duration, means they are providing the duration? quoteType = instalmentAmount, they are providing instalmentAmount?

Comment (Jxr227) [reply src/test/resources/features/ifs/instalment_calculation/InstalmemtCalculationSingleDebt.feature]: If we need to switch these around we can do so - worth validating with Helen I guess?

Comment (byteray) [reply src/test/resources/features/ifs/instalment_calculation/InstalmemtCalculationSingleDebt.feature]: @Jxr227 @rogerball yes let's to avoid any confusion. Happy to have one now quickly. Think she finishes at 4pm

Comment (Jxr227) [reply src/test/resources/features/ifs/instalment_calculation/InstalmemtCalculationSingleDebt.feature]: Helen has agreed @rogerball  my bad for misunderstanding. Will update the code now and the ATs will similarly need to switch around. Great spot!

**Review by Jxr227 — APPROVED** [review]

**Review by thadeusssewanyana — APPROVED** [review]
